File: Chapter 1/Problem 8/Ch1_Problem8.java

public class Ch1_Problem8 {

    public static void printTruthTable(int n, String s) { //printTruthTable( , "");
        if (n == 1) {
            System.out.println(s + "0");
            System.out.println(s + "1");
        } else {
            printTruthTable((n - 1), (s + "0 "));
            printTruthTable((n - 1), (s + "1 "));
        }
    }
}

File: Chapter 1/Problem 8/Ch1_Problem8.cpp
#include <iostream>
using namespace std;

void printTruthTable(int n, string s)
{ // printTruthTable( , "");
    if (n == 1)
    {
        cout << s + "0\n";
        cout << s + "1\n";
    }
    else
    {
        printTruthTable((n - 1), (s + "0 "));
        printTruthTable((n - 1), (s + "1 "));
    }
}